corgi/api/filters.py

```diff
@@ -42,7 +42,7 @@ def filter(self, queryset: QuerySet, value: str) -> QuerySet:
                 )
             else:
                 queryset = queryset.filter(tags__name__icontains=tag)
-        return queryset
+        return queryset.distinct()
```

Comment:
Let's add the .distinct() to the exact part of the queryset that causes issues / multiple results to appear:
```suggestion
                queryset = queryset.filter(tags__name__icontains=tag).distinct()
        return queryset
```
Let's also add a test that confirms this fixes the issue. I think based on the PR description, you're saying that the `__name__icontains` filter returns multiple results for the same component, right? And this happens when any component has multiple tag values with the same tag name, like `{"tag_name": "value_one", "tag_name": "value_two"}`?

Reply:
thx will change distinct location, great advice!